# Preguntas tipo test

**1. Cal é a saída do seguinte fragmento de código?**


In [1]:
x = 1
y = 2
z = 3
z += x
z += y
print (x, y, z)

1 2 6


a) 1 2 3

b) 1 2 6

c) 3 5 3

d) Ningunha das anteriores é correcta

**2. Cal é a saída do seguinte fragmento de código?**

In [2]:
a = 9
b = 2
c = a // b
print(c)

4


a) 5

b) 4.5

c) 4

d) Ningunha das anteriores é correcta

**3. Cal é a saída do seguinte fragmento de código?**

In [3]:
x = "*"
y = 10
z = x * y
print(z)

**********


a) A execución produce un erro

b) `**********`

c) `* * * * * * * * * *`

d) `*10`

**4. Cal é a saída do seguinte fragmento de código?**

In [4]:
nums = (1, 2, 3)
vals = nums
vals[1] = 4
print(nums, vals)

TypeError: 'tuple' object does not support item assignment

a) `[1, 4, 3] [1, 4, 3]`

b) `[1, 2, 3] [1, 4, 3]`

c) `[1, 4, 3] [1, 2, 3]`

d) Ningunha das anteriores é correcta

**5. Cal é a saída do seguinte fragmento de código?**

In [5]:
for i in range(1, 10):
    print(i, end=" ")
    if i % 3 == 0:
        break

1 2 3 

a) 1 2 3 4 5 6 7 8 9 10

b) 1 2 3 4 5 6 7 8 9

c) 1 2

d) 1 2 3

**6. ¿Qué valor se imprime por pantalla?**

In [6]:
def actualizar_saldo(saldo, cantidad, operacion = "sumar"):
    if operacion == "sumar":
        saldo += cantidad
    elif operacion == "restar":
        saldo -= cantidad
    return saldo

print(actualizar_saldo(100, 50))

150


a) 100

b) 50

c) 150

d) Ningunha das anteriores é correcta

**7. Que se imprime por pantalla ao executar o seguinte fragmento de código?**

In [7]:
lista = ["a", "b", "c"]
lista.reverse()
print(lista)

['c', 'b', 'a']


a) `['c', 'b', 'a']`

b) `['a', 'b', 'c']`

c) `['a', 'c', 'b']`

d) Ningunha das anteriores é correcta

**8. Que se imprime por pantalla ao executar o seguinte fragmento de código?**

In [8]:
cadena = "Hola Mundo"
cadena.upper()
print(cadena)

Hola Mundo


a) `Hola Mundo`

b) `hola mundo`

c) `HOLA MUNDO`

d) `Hola mundo`

**9. Que se imprime por pantalla ao executar o seguinte fragmento de código?**

In [9]:
ciudades = ["Madrid", "Valencia", "Barcelona", "Sevilla"]
ciudades.append("Zaragoza")
print(ciudades)

['Madrid', 'Valencia', 'Barcelona', 'Sevilla', 'Zaragoza']


a) `['Madrid', 'Valencia', 'Barcelona', 'Sevilla']`

b) `['Madrid', 'Valencia', 'Barcelona', 'Sevilla', 'Zaragoza']`

c) `['Zaragoza', 'Madrid', 'Valencia', 'Barcelona', 'Sevilla']`

d) Prodúcese un erro

&nbsp;

&nbsp;

**10. Que se imprime por pantalla ao executar o seguinte fragmento de código?**

In [10]:
contador = 0
while contador < 10:
    if contador % 2 == 1:
        continue
    print(contador, end = " ")
    contador += 1

0 

KeyboardInterrupt: 

a) `2 4 8 10`

b) `2 4 8`

c) `1 2 3 4 5 6 7 8 9 10`

d) Ningunha das anteriores é correcta

# Preguntas curtas

**11. Que imprime por pantalla o seguinte fragmento de código?**

In [11]:
vogais = "aeiou"

cadea = "Hola Mundo"

for letra in cadea:
    if letra not in vogais:
        print(letra, end = "")

Hl Mnd

&nbsp;

&nbsp;


**12. Escribe, ao lado de cada liña, que se imprime en pantalla ao executala.**

In [12]:
letras = ["a", "b", "c", "d", "e", "f"]

print(letras[:1])

print(letras[-2])

print(letras[1:4:2])

print(letras[3:])

print(letras[::-1])

['a']
e
['b', 'd']
['d', 'e', 'f']
['f', 'e', 'd', 'c', 'b', 'a']




**13. Emprega comprensión de listas para que a variable `filtrada` conteña só os elementos pares da lista `numeros`**

In [15]:
numeros = [4, 6, 3, 7, 9, 10, 1]
#filtrada =

In [17]:
filtrada = [i for i in numeros if i % 2 == 0]
print(filtrada)

[4, 6, 10]


&nbsp;

&nbsp;

&nbsp;

&nbsp;

&nbsp;


**14. Convirte este bucle `for` para que a lista se percorra por índices e non por valores.**

In [ ]:
lista = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for valor in lista:
    print(valor)

In [18]:
lista = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for i in range(len(lista)):
    print(lista[i])

1
2
3
4
5
6
7
8
9
10


&nbsp;

&nbsp;

&nbsp;

&nbsp;

&nbsp;

**15. Que se imprime por pantalla ao executar o seguinte fragmento de código?**

In [19]:
for i in range(3):
    for j in range(3):
        if i == j:
            print("x", end="")
        else:
            print("o", end="")
    print()

xoo
oxo
oox
